astropy/visualization/wcsaxes/tests/test_wcsapi.py

```diff
@@ -266,6 +267,37 @@ def test_coord_type_from_ctype(cube_wcs):
     assert coord_meta["format_unit"] == [u.one, u.one]
     assert coord_meta["wrap"] == [None, None]
 
+    myframe_mapping = {
+        "custom:pos.myframe.lon": {
+            "coord_wrap": 180.0 * u.deg,
+            "format_unit": u.arcsec,
+            "coord_type": "longitude",
+        },
+        "custom:pos.myframe.lat": {"format_unit": u.arcsec, "coord_type": "latitude"},
+    }
+
+    astropy.visualization.wcsaxes.wcsapi.CUSTOM_UCD_COORD_META_MAPPING.update(
+        myframe_mapping
+    )
```

Comment:
I think we should avoid having this be the API to register custom mappings, and instead write a context manager similar to ``custom_ctype_to_ucd_mapping`` to make sure the changes are temporary and cleaned up after use.

Reply:
This would also have the benefit of allowing this functionality to be documented via the API docs even if we don't add something to the main narrative docs.

Reply:
Great thanks for the feedback, I've taken a first crack at adding a context manager similar `custom_ctype_to_ucd_mapping`.

Reply:
Thanks! I don't think we should keep the test of modifying the dictionary, we should just be testing the context manager.

Reply:
Given there's no way to stop someone editing the dictionary shouldn't we keep a test for it?

Reply:
I'd vote against it; we shouldn't be testing for implementation details.

Reply:
If the dictionary isn't meant to be accessed directly maybe we should make it private to make it explicit?

Reply:
I removed it, I don't think we need to make it private

Reply:
I do think there is utility in people directly accessing the dict if packages like sunpy write their own coordinate frames they could add to the dict without the context manager to improve the plotting of those frames.